src/volatile_memory.rs

```diff
@@ -285,10 +336,16 @@ impl<'a, B: BitmapSlice> VolatileSlice<'a, B> {
     /// Returns a pointer to the beginning of the slice. Mutable accesses performed
     /// using the resulting pointer are not automatically accounted for by the dirty bitmap
     /// tracking functionality.
+    #[cfg(not(all(feature = "xen", unix)))]
     pub fn as_ptr(&self) -> *mut u8 {
         self.addr
     }
 
+    /// Returns a guard for the pointer to the underlying memory.
+    pub fn as_ptr_guard(&self, is_read: bool) -> VolatileSliceGuard {
```

Comment:
I am trying to compare this to other APIs that I have seen within the standard library... Usually `as_*` are just simpler conversion methods that have close to 0 overhead, right? Would it maybe make sense to just call this `(m)map()`?

Reply:
Named this as ptr_guard() and ptr_guard_mut() now.

Reply:
LGTM (cannot mark as resolved since I commented on a commit that no longer exists)